cdap-api/src/main/java/co/cask/cdap/api/dataset/module/DatasetDefinitionRegistry.java

```diff
@@ -34,7 +36,7 @@
    * @param def definition to add
    * @throws IllegalArgumentException if registry already contains dataset type of the same name as given definition
    */
-  void add(DatasetDefinition def);
+  void add(DatasetDefinition def) throws IOException;
```

Comment:
let's not start throwing IOException from the dataset framework. This is written in a way that it propagates (almost) everything as RuntimeException. We throw IOException for system tables, but let's try not to change the existing dataset framework too much.